src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -899,6 +923,44 @@ protected static int GetParameterIndex<TNode>(SeparatedSyntaxList<TNode> paramet
             return Generator.SeparatedList(fullList.ToImmutableAndFree(), separators.ToImmutableAndFree());
         }
 
+        private async Task<(SyntaxNode expression, bool callsiteInferenceFailed)> GenerateInferredExpressionAsync(
+            SemanticModel semanticModel,
+            Workspace workspace,
+            int position,
+            ITypeSymbol targetType,
+            CancellationToken cancellationToken)
+        {
+            var recommendations = await Recommender.GetImmutableRecommendedSymbolsAtPositionAsync(
+                semanticModel, position, workspace, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+            var sourceSymbols = recommendations.Where(r => r.IsNonImplicitAndFromSource());
+            var orderedLocalAndParameterSymbols = sourceSymbols
+                .Where(s => s.IsKind(SymbolKind.Local) || s.IsKind(SymbolKind.Parameter))
+                .OrderByDescending(s => s.Locations.First().SourceSpan.Start);
```

Comment:
same issue arises about this being super weird wrt to partial types.

Reply:
you might want a comment.  i suppose it won't matter though because locals/parameers have to be in teh same context as you.  

Note: you also want to filter to SymbolKind.RangeVariable.

Reply:
this can pick a variable defined *after* the callsite.  taht seems not good.  i think we should ensure it's before the callsite.  personally i think teh closer to the callsite the better.  but i'm ok with you not doing that step.

Reply:
Okay. I filed [an issue](https://github.com/dotnet/roslyn/issues/44689) for range variables and no longer attempt to sort fields/properties. We are prevented from picking a local defined after the callsite by the Recommender, but I've added a test to ensure this.